nova/virt/vmwareapi/vmops.py

```diff
@@ -3306,3 +3401,53 @@ def _sync_sync_server_group(context, sg_uuid):
             LOG.debug('Sync for server-group %s done', sg_uuid)
 
         _sync_sync_server_group(context, sg_uuid)
+
+    def place_vm(self, context, instance):
+        # We currently only fill the bare-minimum to get a placement.
+        # The datastore for the VM is selected as on instance creation,
+        # instead of allowing placevm to decide it (which may be better)
+        # We also do not pass the information about the NICs and other
+        # attached disks, which may give the placement a better information
+        client_factory = self._session.vim.client.factory
+        flavor = instance.flavor
+        image_meta = instance.image_meta
+        image_info = images.VMwareImage.from_image(context,
+                                                instance.image_ref,
+                                                image_meta)
+
+        extra_specs = self._get_extra_specs(flavor, image_meta)
+
+        vi = self._get_vm_config_info(instance, image_info, extra_specs)
+
+        vm_folder = self._get_project_folder(vi.dc_info,
+            project_id=instance.project_id, type_='Instances')
+
+        relocate_spec = vm_util.relocate_vm_spec(
+            client_factory,
+            res_pool=self._root_resource_pool,
+            datastore=vi.datastore.ref,
+            disk_move_type="moveAllDiskBackingsAndDisallowSharing",
+            folder=vm_folder)
+
+        placement_spec = client_factory.create("ns0:PlacementSpec")
+        placement_spec.placementType = "relocate"
+        # Maybe we want to allow that? Default is True
+        # placement_spec.disallowPrerequisiteMoves = False
+        placement_spec.relocateSpec = relocate_spec
+        # So we do not place the vm on a failover host
+        placement_spec.hosts, _ = \
+            vm_util.get_hosts_and_reservations_for_cluster(
+                self._session, self._cluster)
+
+        # Sets cpuAllocation, memoryAllocation, numCPUs, memoryMB
+        config_spec = vm_util.get_vm_resize_spec(client_factory,
+                                                    int(flavor.vcpus),
+                                                    int(flavor.memory_mb),
+                                                    extra_specs)
+        # The last mandatory field for config_spec per doc
+        config_spec.version = extra_specs.hw_version
```

Comment:
> Could VMware by that decide to place it on a host which does not support our (probably older) version?
I think, that is a rather theoretical situation. To my knowledge, newer ESXi versions support all oder versions.
It is also written in the UI that way: `ESXi version X.Y or later`.

The way I see it, we only risk a problem, if we switch versions forward in the config to a version not supported in part of our cluster, create vms with that config, and then move the config back to an older version. 
But since we only can change the version for a whole region, we probably won't do that.

> Wouldn't it be safer to use the VMs hardware version?

I think getting it from the VM is the better solution. It would be more self-contained, and less dependent on out operative procedures. 
The main reason for me avoiding it, is the coding effort. I do not see it really being needed for our use-case.